### PR TITLE
[browser][MT] preparation for deputy

### PR DESCRIPTION
**Pull request by pavelsavara**
- rename `JSProxyContext.NativeTID` to `JSNativeTID`
- pass TID to `InstallMainSynchronizationContext` and `GCHandle` back
- don't use synchronous resolve for `Task` pre-resolved on another thread
- fix CTS leak in `WaitAssertsOnJSInteropThreads`
- rename `mono_wasm_invoke_method_bound` to `mono_wasm_invoke_method`
- drop `mono_wasm_invoke_method_raw` in favor of `mono_wasm_invoke_method`
- load `segmentation-rules` and `symbols` assets into worker
- normalized functions in `managed-exports.ts`
- renamed `runtimeHelpers.proxyGCHandle`
- handle exit_code === undefined
- handle document undefined in `appendElementOnExit` in worker
- made `loaderHelpers.invariantMode` local variable
- improved `thread_dump` for detached threads and UI thread
- `mono_wasm_init_threads` asynchronously and earlier and moved the code
- moved more logic into `start_runtime`
- rename `invoke_method_and_handle_exception` to `invoke_sync_method`
- fix `RuntimeHelpers` init of `nativeExit` and `nativeAbort`

### CODE REVIEW

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs [hunk 1]
```diff
@@ -296,11 +296,11 @@ public static void GetManagedStackTrace(JSMarshalerArgument* arguments_buffer)
...
         [DynamicDependency(DynamicallyAccessedMemberTypes.NonPublicMethods, "System.Runtime.InteropServices.JavaScript.JSWebWorker", "System.Runtime.InteropServices.JavaScript")]
-        // the marshaled signature is:
-        // void InstallMainSynchronizationContext()
-        public static void InstallMainSynchronizationContext()
+        public static void InstallMainSynchronizationContext(nint* nativeTIDAndProxyContextGCHandlePtr)
         {
-            JSSynchronizationContext.InstallWebWorkerInterop(true, CancellationToken.None);
+            var jsSynchronizationContext = JSSynchronizationContext.InstallWebWorkerInterop(true, CancellationToken.None);
+            jsSynchronizationContext.ProxyContext.JSNativeTID = *nativeTIDAndProxyContextGCHandlePtr;
+            *nativeTIDAndProxyContextGCHandlePtr = jsSynchronizationContext.ProxyContext.ContextHandle;
```
Comment (maraf) [inline src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs:303]: This is hard, but understandable in the name of performance I guess 🤷‍♂️

**Review by maraf — APPROVED** [review]
I did not spot anything ungood

**Review by ilonatommy — APPROVED** [review]

File: src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs (hunk 1 above)
Comment (pavelsavara) [reply src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs]: This was not about perf. I updated it to use more established pattern.

**Pull request merged**